### PR TITLE
nixos/qt: extend

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

See #71079
Gtk part #80865

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bqv** [comment]
@worldofpeace @ttuegel ping

**Review by ttuegel — APPROVED** [review]
Looks good to me!

File: nixos/modules/config/qt.nix [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        {
+          assertion = cfg.font == null && cfg.iconTheme == null;
+          message = "`qt.font` and `qt.iconTheme` are only supported by kde platform.";
+        }
+      ];
+      environment.variables.QT_QPA_PLATFORMTHEME = "qgnomeplatform";
+      # TODO: make this optional once https://github.com/NixOS/nixpkgs/issues/54150 is fixed
+      # qgnomeplatform reads theme and other settings from dconf db
+      environment.variables.QT_STYLE_OVERRIDE = cfg.style.name;
+      environment.variables.XDG_DATA_DIRS = [ "${pkgs.gsettings-desktop-schemas}/share/gsettings-schemas/${pkgs.gsettings-desktop-schemas.name}" ];
```
Comment (worldofpeace) [inline nixos/modules/config/qt.nix]:
```suggestion
```

I'm going to merge https://github.com/NixOS/nixpkgs/pull/83245, so we won't need this.
It's also a bad idea to set XDG_DATA_DIRS in general like this.

File: nixos/modules/services/x11/desktop-managers/pantheon.nix [hunk 2]
```diff
@@ -242,9 +242,9 @@ in
       programs.zsh.vteIntegration = mkDefault true;
 
       # Harmonize Qt5 applications under Pantheon
-      qt5.enable = true;
-      qt5.platformTheme = "gnome";
-      qt5.style = "adwaita";
+      qt.enable = true;
+      qt.platformTheme = "qgnomeplatform";
+      qt.style.name = "adwaita";
```
Comment (worldofpeace) [inline nixos/modules/services/x11/desktop-managers/pantheon.nix]: You're missing the adwaita-qt theme package.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Forgot to request changes

**Comment by gnidorah** [comment]
@worldofpeace Fixed, thank you!

**Comment by Quantenzitrone** [comment]
yo what happened to this?

**Comment by Quantenzitrone** [comment]
yo what happened to this?

**Pull request closed without merging**